core/src/main/java/org/testcontainers/containers/wait/LogMessageWaitStrategy.java

```diff
@@ -0,0 +1,37 @@
+package org.testcontainers.containers.wait;
+
+import org.testcontainers.containers.ContainerLaunchException;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.OutputFrame;
+import org.testcontainers.containers.output.WaitingConsumer;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Predicate;
+
+/**
+ * Waits until containers logs expected content.
+ */
+public class LogMessageWaitStrategy extends GenericContainer.AbstractWaitStrategy {
+    private String expectedLogPart;
+
+    @Override
+    protected void waitUntilReady() {
+        WaitingConsumer waitingConsumer = new WaitingConsumer();
+        container.followOutput(waitingConsumer);
+
+        Predicate<OutputFrame> waitPredicate = outputFrame ->
+                outputFrame.getUtf8String().contains(expectedLogPart);
```

Comment:
I'd imagine an overloaded `withExpectedLogPart` method that accepts a `Pattern` could be flexible enough, couldn't it?